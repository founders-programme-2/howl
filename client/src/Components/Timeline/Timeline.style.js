import styled from 'styled-components';

const Picture = styled.img`
  width: 70%;
  margin: 2rem auto;
`;

const ImgFigure = styled.figure`
  width: 100%;
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  margin: 2rem auto;
`;

const FigCap = styled.figcaption`
  width: 80%;
  margin: 0 auto;
`;

export { Picture, ImgFigure, FigCap };
