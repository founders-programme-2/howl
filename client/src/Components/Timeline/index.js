import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import Footer from '../Common/Footer';
import navigationUrls from '../../constants/navigationUrls';
import { Picture, ImgFigure, FigCap, TimelineHeader } from './Timeline.style';
import TimelinePic from '../../assets/TimelinePic.svg';
import SelectTimeFrame from '../../assets/SelectTimeFrame.svg';
import CategorySelected from '../../assets/CategorySelected.svg';
import TagsSelected from '../../assets/TagsSelected.svg';
import SelectedMultipleFilters from '../../assets/SelectedMultipleFilters.svg';
import SearchFunctionality from '../../assets/SearchFunctionality.svg';
import StoriesHover from '../../assets/StoriesHover.svg';

const Timeline = () => {
  const { ARCHIVE_URL } = navigationUrls;
  return (
    <Fragment>
      <main>
        <header>
          <TimelineHeader>
            The HOWL Timeline -
            <br />
            Coming soon!
          </TimelineHeader>
          <ImgFigure>
            <Picture src={TimelinePic} alt="Coming soon interactive timeline" />
            <FigCap>Coming soon interactive timeline</FigCap>
          </ImgFigure>
        </header>
        <h2>How it works</h2>
        <p>
          Each hexagon represents a story/event. You can hover on each story to
          enlarge it and get a snippet of when the story took place, the title
          and a small link that you can click to read more.
        </p>
        <ImgFigure>
          <Picture
            src={StoriesHover}
            alt="A picture depicting a hovered enlarged story"
          />
          <FigCap>A picture depicting a hovered enlarged story</FigCap>
        </ImgFigure>
        <p>
          Browse the timeline by using the controls at the bottom of the page to
          filter your results by category, location, tags or year. Above the
          search controls, you&#39;ll find a yellow bar of dates. This allows
          you to view stories from between a time span of four to five years.
          For example, If you click on the first time frame (1960-1965) it will
          zoom in on that time frame only. Allowing you a clearer view of events
          and when they happened.
        </p>
        <ImgFigure>
          <Picture
            src={SelectTimeFrame}
            alt="A picture depicting the selected time frame (1960-1965) stories"
          />
          <FigCap>
            A picture depicting the selected time frame (1960-1965) stories
          </FigCap>
        </ImgFigure>
        <p>
          You can use the controls in the Footer of the page to filter the
          timeline and focus on events relevant to your chosen topic. For
          example, you can filter by single select dropdowns like category.
        </p>
        <ImgFigure>
          <Picture
            src={CategorySelected}
            alt="A picture depicting stories filtered by selected category 'major event'"
          />
          <FigCap>
            {
              "A picture depicting stories filtered by selected category 'major event'"
            }
          </FigCap>
        </ImgFigure>
        <p>{"You can also apply multiselect filters like 'Tags'."} </p>
        <ImgFigure>
          <Picture
            src={TagsSelected}
            alt="A picture depicting stories filtered by selected tags: abortion, reproductive rights, conferences & childcare."
          />
          <FigCap>
            A picture depicting stories filtered by selected Tags: abortion,
            reproductive rights, conferences & childcare.
          </FigCap>
        </ImgFigure>
        <p>
          The more filters you apply, the more you will be able to narrow down
          your search and customize stories displayed on the timeline.
        </p>
        <ImgFigure>
          <Picture
            src={SelectedMultipleFilters}
            alt="A picture depicting stories filtered by both selected category 'story' and selected tags: abortion, reproductive rights, conferences & childcare."
          />
          <FigCap>
            {
              "A picture depicting stories filtered by both selected category 'story' and selected tags: abortion, reproductive rights, conferences & childcare."
            }
          </FigCap>
        </ImgFigure>
        <p>
          Looking for very specific keywords? No worries! You can use the search
          functionality to filter through stories that either match or contain
          whatever word you enter in the search field.
        </p>
        <ImgFigure>
          <Picture
            src={SearchFunctionality}
            alt="A picture depicting stories filtered by searching for keyword 'abortion'"
          />
          <FigCap>
            {
              "A picture depicting stories filtered by searching for keyword 'abortion'"
            }
          </FigCap>
        </ImgFigure>

        <p>
          Prefer a simpler approach? Take a look at the{' '}
          <Link to={ARCHIVE_URL}>archive</Link> for a text-based experience.
        </p>
      </main>
      <Footer />
    </Fragment>
  );
};

export default Timeline;
