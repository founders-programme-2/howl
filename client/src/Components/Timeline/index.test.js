import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Timeline from './index';

test('Natural', () => {
  const timeline = renderer
    .create(
      <BrowserRouter>
        <Timeline />{' '}
      </BrowserRouter>
    )
    .toJSON();
  expect(timeline).toMatchSnapshot();
});
