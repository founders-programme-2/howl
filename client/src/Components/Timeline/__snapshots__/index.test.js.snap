// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Natural 1`] = `
Array [
  <main>
    <header>
      <h1>
        The HOWL Timeline
      </h1>
    </header>
    <p>
      Browse our timeline by using the controls at the bottom of this page to filter your results by category, location, tags or year. Above the search controls, you'll find a yellow bar of dates. This allows you to view stories from between a time span of four to five years.
    </p>
    <p>
      Prefer a simpler approach? Take a look at the
       
      <a
        href="/archive"
        onClick={[Function]}
      >
        archive
      </a>
       for a text-based experience.
    </p>
  </main>,
  <footer
    className="sc-bdVaJa dycKuB"
  >
    <section
      className="sc-bwzfXH fiRAvd"
    >
      <h2
        className="sc-htpNat gKOkZF"
      >
        Search Archive
      </h2>
      <button
        className="MuiButtonBase-root MuiButton-root Footer-howToSearchButton-1 MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          How to search
        </span>
      </button>
    </section>
    <section
      className="sc-bxivhb jbVdgm"
    >
      <div
        className="MuiFormControl-root Footer-formControl-2"
      >
        <div
          aria-label="category"
          className="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root-44 MuiInputBase-formControl Footer-select-7"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-45"
              id="select-category"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="sc-bZQynM kihKbR"
              >
                Category
              </span>
            </div>
            <input
              id="category-customized-select"
              name="category"
              type="hidden"
              value="Category"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <div
          aria-label="location"
          className="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root-44 MuiInputBase-formControl Footer-select-7"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-45"
              id="select-location"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="sc-bZQynM kihKbR"
              >
                Location
              </span>
            </div>
            <input
              id="location-customized-select"
              name="location"
              type="hidden"
              value="Location"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <div
          aria-label="year"
          className="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root-44 MuiInputBase-formControl Footer-select-7"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-45"
              id="select-year"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="sc-bZQynM kihKbR"
              >
                Year
              </span>
            </div>
            <input
              id="year-customized-select"
              name="year"
              type="hidden"
              value="Year"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root Footer-multipleSelectFormControl-3"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Footer-MultipleSelectInputLabel-6"
          data-shrink={false}
          htmlFor="selectMultipleCheckbox"
        >
          Tags
        </label>
        <div
          aria-label="tags"
          className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiInput-input"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              id="selectMultipleCheckbox"
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </section>
    <section
      className="sc-ifAKCX ezvIgM"
    >
      <div
        className="MuiFormControl-root Footer-searchFormControl-8"
      >
        <div
          className="MuiFormControl-root MuiTextField-root Footer-searchTextField-9"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Footer-searchTextFieldLabel-10"
            data-shrink={false}
            htmlFor="outlined-search"
          >
            Search...
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
              disabled={false}
              id="outlined-search"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="search"
              value=""
            />
          </div>
        </div>
      </div>
      <section
        className="sc-EHOje khXZcR"
      >
        <button
          className="MuiButtonBase-root MuiButton-root Footer-searchButton-11 MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Search
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root Footer-resetButton-12 MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Reset
          </span>
        </button>
      </section>
    </section>
  </footer>,
  " ",
]
`;
